### PR TITLE
Tom/DLS-5487 [CBCR] Play 2.8.16 upgrade for cbcr-frontend

### PR DESCRIPTION
**Pull request by TomRafferty**
**Changes**
- Updated as many dependencies to the latest version as I could without codebase changes (out of scope) this includes play and bootstrap.

- Updated tests to reflect the changes to the libraries.

- Updated copyright year (2022 -> 2023)

All unit tests pass
Help from Michael Wolfendale

### CODE REVIEW

File: app/uk/gov/hmrc/cbcrfrontend/controllers/ExitSurveyController.scala [hunk 1]
```diff
@@ -31,7 +33,7 @@ import uk.gov.hmrc.play.audit.http.connector.{AuditConnector, AuditResult}
 import uk.gov.hmrc.play.audit.model.ExtendedDataEvent
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.{Await, ExecutionContext, Future}
```
Comment (LAJW) [inline app/uk/gov/hmrc/cbcrfrontend/controllers/ExitSurveyController.scala:36]: ⛏️ Unused import

File: app/uk/gov/hmrc/cbcrfrontend/controllers/ExitSurveyController.scala [hunk 2]
```diff
@@ -56,14 +58,15 @@ class ExitSurveyController @Inject()(
...
+        answers => {
           auditSurveyAnswers(answers).fold(
             errors => {
               logger.error(errors.toString)
               Redirect(routes.ExitSurveyController.surveyAcknowledge)
             },
             _ => Redirect(routes.ExitSurveyController.surveyAcknowledge)
-        )
+          )
+        }
```
Comment (LAJW) [inline app/uk/gov/hmrc/cbcrfrontend/controllers/ExitSurveyController.scala:69]: ⛏️ No-op changes, you can revert all changes in this file.

File: app/uk/gov/hmrc/cbcrfrontend/controllers/StartController.scala [hunk 3]
```diff
@@ -65,17 +66,24 @@ class StartController @Inject()(
...
+                Redirect(routes.FileUploadController.chooseXMLFile)
+              case "editSubscriberInfo" => {
+                Redirect(routes.SubscriptionController.updateInfoSubscriber)
+              }
+              case _ => {
+                BadRequest(views.start(startForm))
+              }
+            }
+          )
+      }
```
Comment (LAJW) [inline app/uk/gov/hmrc/cbcrfrontend/controllers/StartController.scala:86]: ⛏️ Only whitespace changes, can be reverted.

File: app/uk/gov/hmrc/cbcrfrontend/controllers/SubscriptionController.scala [hunk 4]
```diff
@@ -175,39 +175,41 @@ class SubscriptionController @Inject()(
...
+              _ <- subscriptionDataService.updateSubscriptionData(
+                    cbcId,
+                    SubscriberContact(data.firstName, data.lastName, data.phoneNumber, data.email))
+            } yield Redirect(routes.SubscriptionController.savedUpdatedInfoSubscriber)).fold(
+              errors => errorRedirect(errors, views.notAuthorisedIndividual, views.errorTemplate),
+              result => result
+            )
+          }
+        )
+      }
```
Comment (LAJW) [inline app/uk/gov/hmrc/cbcrfrontend/controllers/SubscriptionController.scala:212]: ⛏️ Was this autoformatted? The only change is adding `Action.async(parse.formUrlEncoded)` at the start, diff could be reduced if it wasn't autoformatted.

**Review by LAJW — DISMISSED** [review]
Could be merged as is. If you want, you can try to remove some of the whitespace-only changes.

File: project/plugins.sbt [hunk 5]
```diff
@@ -4,11 +4,11 @@ resolvers += "HMRC-open-artefacts-maven" at "https://open.artefacts.tax.service.
 resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts.tax.service.gov.uk/ivy2"))(Resolver.ivyStylePatterns)
 resolvers += Resolver.jcenterRepo
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.3.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "3.8.0")
 
 addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.1.0")
 
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.8" exclude("org.slf4j", "slf4j-simple"))
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.15" exclude("org.slf4j", "slf4j-simple"))
```
Comment (rambabu-posa) [inline project/plugins.sbt:11]: Hi @TomRafferty  Please update this to 2.8.18 (All Teams are agreed to use this version)

**Review by LAJW — APPROVED** [review]

**Review by LAJW — APPROVED** [review]

File: project/plugins.sbt (hunk 5 above)
Comment (TomRafferty) [reply project/plugins.sbt]: done :) thanks

**Pull request merged**